rig-mongodb/tests/integration_tests.rs

```diff
@@ -92,6 +93,65 @@ async fn vector_search_test() {
     )
 }
 
+async fn create_search_index(collection: &Collection<bson::Document>) {
+    let max_attempts = 5;
+
+    for attempt in 0..max_attempts {
+        match collection
+            .create_search_index(
+                SearchIndexModel::builder()
+                    .name(Some(VECTOR_SEARCH_INDEX_NAME.to_string()))
+                    .index_type(Some(mongodb::SearchIndexType::VectorSearch))
+                    .definition(doc! {
+                        "fields": [{
+                            "numDimensions": 1536,
+                            "path": "embedding",
+                            "similarity": "cosine",
+                            "type": "vector"
+                        }]
+                    })
+                    .build(),
+            )
+            .await
+        {
+            Ok(_) => {
+                // Wait for index to be available
+                for _ in 0..max_attempts {
+                    let indexes = collection
+                        .list_search_indexes()
+                        .await
+                        .unwrap()
+                        .collect::<Vec<_>>()
+                        .await;
```

Comment:
Tip: Instead of fetching all indices, you can filter by name and get only one:
```rust
let maybe_index = collection
    .list_search_indexes()
    .name(VECTOR_SEARCH_INDEX_NAME)
    .await
    .unwrap()
    .next()
    .await
```